File: CS50/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string arr)
{
    int text_length = strlen(arr);
    int num=0;
    for(int i=0; i<text_length;i++)
    {
        if (isalpha(arr[i])){
            num +=1;
        }
    }
    return num;
}

int count_words(string arr)
{
    int text_length = strlen(arr);
    int space=1;
    for(int i=0; i<text_length;i++)
    {
        if (isspace(arr[i])){
            space +=1;
        }
    }
    return space;
}
int count_sentences(string arr)
{
    int text_length = strlen(arr);
    int periods=0;
    for(int i=0; i<text_length;i++)
    {
        if (arr[i]=='?' || arr[i]=='!' || arr[i]=='.'){
            periods +=1;
        }
    }
    return periods;
}

float cli(int l,int w ,int s){

    float L= l*100/w;
    float S= s*100/w;
    float index= 0.0588 * L - 0.296 * S - 15.8;
    return index;
}
int main(void)
{
    string text = get_string("Text:");
    int letters =count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    printf("Letters: %i \n", letters);
    printf("Words: %i \n", words);
    printf("Sentences: %i \n", sentences);
    int index= round(cli(letters, words, sentences));
    if (index < 1){
        printf("Before Grade 1\n");
    }else if (index>16){
        printf("Grade 16+\n");
    }else{
        printf("Grade %i\n",index);
    }
}

File: CS50/pset6/readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")
    letters=count_letters(text)
    words=count_words(text)
    sentences=count_sentences(text)
    print("Letters:",letters)
    print("Words:",words)
    print("Sentences",sentences)
    index= int(cli(letters,words,sentences))
    if index<1:
        print("Before Grade 1")
    elif index>16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

def count_letters(string):
    text_length = len(string)
    letters=1
    for i in range(text_length):
        if string[i].isalpha():
            letters +=1
    return letters

def count_words(string):
    text_length = len(string)
    space=1
    for i in range(text_length):
        if string[i]==' ':
            space +=1
    return space

def count_sentences(string):
    text_length = len(string)
    periods=0
    for i in range(text_length):
        if string[i] in ['?','!','.']:
            periods +=1
    return periods

def cli(l,w,s):
    L=l*100/w
    S=s*100/w
    index=0.0588 * L - 0.296 * S - 15.8
    return index

if __name__=='__main__':
    main()